ontology/yaml/resources/SAFETY/entity_types/ABSTRACT.yaml

```diff
@@ -79,18 +79,26 @@ FA2X:
   uses:
   - fire_alarm_1
   - fire_alarm_2
+  opt_uses:
+  - primary_fire_alarm
+  - secondary_fire_alarm
     
 WLS:
   guid: "16a99ce4-3298-4f9a-9ba8-41edec6a48c8"
   description: "Water leak location monitoring."
   is_abstract: true
   uses:
   - water_leak_cable_distance_sensor
-  
+
 LM:
   guid: "25887ceb-f2d8-4b25-be80-36e8b37f5ae9"
   description: "Tank level monitoring."
   is_abstract: true
   uses:
-  - level_status 
+  - level_status
 
+AL:
```

Comment:
Does not define fields, so this is not a well defined abastract type. Either add the alarms you need here, or directly to the canonical entities.

Reply:
noted sir, give me a few days to review all the needed changes for all of what you noted before I push commits. Thanks sir.